app/src/chainlit.py

```diff
@@ -31,14 +35,33 @@ async def start() -> None:
         ).send()
         return
 
-    settings = await _init_chat_settings(engine)
+    input_widgets = _init_chat_settings(engine, query_values)
+    settings = await cl.ChatSettings(input_widgets).send()
+    logger.info("Initialized settings: %s", pprint.pformat(settings, indent=4))
+    if query_values:
```

Comment:
I hadn't seen this pattern with pop and then checking for remaining before reviewing your PRs, it's really slick!